pkg/spec/job.go

```diff
@@ -115,6 +115,12 @@ func (job *JobSpecMod) createKubernetesController() (*batch_v1.Job, error) {
 	if job.ActiveDeadlineSeconds != nil {
 		jobSpec.ActiveDeadlineSeconds = job.ActiveDeadlineSeconds
 	}
+	// if RestartPolicy is not set by user default it to 'OnFailure'
```

Comment:
IMO we should move this logic to `func (job *JobSpecMod) Fix() error {`

We can create something like `func (job *JobSpecMod) fixJob()` like in case of DeploymentConfig -

```golang
func (deploymentConfig *DeploymentConfigSpecMod) Fix() error {

	if err := deploymentConfig.ControllerFields.fixControllerFields(); err != nil {
		return errors.Wrap(err, "unable to fix ControllerFields")
	}

	// Fix DeploymentConfig
	deploymentConfig.fixDeploymentConfig()

	return nil
}

func (deploymentConfig *DeploymentConfigSpecMod) fixDeploymentConfig() {
	deploymentConfig.ControllerFields.ObjectMeta.Labels = addKeyValueToMap(appLabelKey,
		deploymentConfig.ControllerFields.Name,
		deploymentConfig.ControllerFields.ObjectMeta.Labels)
...
...
```

Reply:
Let's do this refactor later on I think!

Reply:
Any specific reason that we do this later? I think this fits in the scope of this PR. All the fix related code should go there, WDYT?